lib/mix/lib/mix/cli.ex

```diff
@@ -43,6 +43,9 @@ defmodule Mix.CLI do
     exit({:shutdown, 1})
   end
 
+  defp get_task([testfile = "test/" <> _|t]) do
+    {"test", [testfile|t]}
+  end
```

Comment:
This does not consider the `:test_paths` configuration and it should probably also match against `:test_pattern`.